### ISSUE TITLE
Rtex code extraction problematic

### ISSUE THREAD

**Issue by AshesITR** [r-lib/lintr#1043]
When bringing #165 up to speed, I noticed a test failure for Rtex, caused by the behavior of `get_source_expressions()`:

```r
get_source_expressions("tests/testthat/knitr_formats/test.Rtex")$lines[[25]]
#> [1] " "
```

The original file line in question is `"%"`, where the `%` is masked by whitespace as part of the Rtex markup.
This results in a false positive `trailing_whitespace_linter()` hit for that line once it's extended to cover all lines and not just lines with expressions.

**Comment by MichaelChirico** [comment]
Copying over some more context from a TODO comment:

> `file_lines` contains a whitespace on the final line for Rtex, because that is used to mark the Rtex escape char "%" as well. c.f. `get_source_expressions("tests/testthat/knitr_formats/test.Rtex")$lines[[25]]`
